### PR TITLE
Fix image measure function to apply inherent aspect ratio to style sizes

### PR DESCRIPTION
**Pull request by nicoburns**
# Objective

- Fixes https://github.com/bevyengine/bevy/issues/13155
- fixes https://github.com/bevyengine/bevy/issues/13517
- Supercedes https://github.com/bevyengine/bevy/pull/13381
- Requires https://github.com/DioxusLabs/taffy/pull/661

## Solution

- Taffy has been updated to:
    - Apply size styles to absolutely positioned children
    - Pass the node's `Style` through to the measure function
- Bevy's image measure function has been updated to make use of this style information

## Notes

- This is currently using a git version of Taffy. If this is tested as fixing the issue then we can turn that into a Taffy 0.5 release (this would be the only change between Taffy 0.4 and Taffy 0.5 so upgrading is not expected to be an issue)
- This implementation may not be completely correct. I would have preferred to extend Taffy's gentest infrastructure to handle images and used that to nail down the correct behaviour. But I don't have time for that atm so we'll have to iterate on this in future. This PR at least puts that under Bevy's control.

## Testing

- I manually tested the game_menu_example (from https://github.com/bevyengine/bevy/issues/13155)
- More testing is probably merited

---

## Changelog

No changelog should be required as it fixes a regression on `main` that was not present in bevy 0.13. The changelog for "Taffy upgrade" may want to be changed from 0.4 to 0.5 if this change gets merged.

### CODE REVIEW

**Comment by nicoburns** [comment]
CC @bugsweeper @porkbrain

**Comment by alice-i-cecile** [comment]
@bugsweeper can I get your review here as well?

File: crates/bevy_ui/src/widget/image.rs
Comment (mockersf) [inline crates/bevy_ui/src/widget/image.rs:74]: not sure I understand what you mean by this comment

**Review by bugsweeper — COMMENTED** [review]
This PR contains changes done according to one of previously approved in the issue ways. Looks solid. Examples work correctly. There is only one problem, in this PR bevy_ui depends on personal fork of taffy, it's the best what we can get from changing two separate but dependent crates at same time, but it's not suitable for release. Should we wait some time (two weeks?) for taffy new release?

**Comment by nicoburns** [comment]
> There is only one problem, in this PR `bevy_ui` depends on personal fork of `taffy`, it's the best what we can get from changing two separate but dependent crates at same time, but it's not suitable for release. Should we wait some time (two weeks?) for taffy new release?

Me (and Alice) are maintainers of Taffy. I/we can put out a release whenever one is needed, I just don't want to put out a release that doesn't actually fix the problem. The fix looks like it works for #13155 to me, but I was hoping that the proposed fix could get a little more testing, and confirmation from at least one other person that it works first.

**Comment by bugsweeper** [comment]
> Me (and Alice) are maintainers of Taffy

Verry well, I didn't know that.

File: crates/bevy_ui/Cargo.toml
Comment (Weibye) [inline crates/bevy_ui/Cargo.toml:34]: I assume the plan is to revert this to 4.x once taffy releases with this changes?

File: crates/bevy_ui/src/widget/image.rs
Comment (Weibye) [inline crates/bevy_ui/src/widget/image.rs:74]:
```suggestion
        // If either width or height was determined at this point, then both are determined beyond this point.
```
Does this phrasing 1) clear up the confusion @mockersf , and 2) correctly convey your meaning @nicoburns ?

Comment (Weibye) [reply crates/bevy_ui/src/widget/image.rs:74]: I agree that the meaning of the last half of this sentence isn't self-evident

**Review by Weibye — APPROVED** [review]

File: crates/bevy_ui/Cargo.toml
Comment (nicoburns) [reply crates/bevy_ui/Cargo.toml:34]: It will bump to 0.5 but yes, the plan is to revert to a crates.io (before this PR is merged).

File: crates/bevy_ui/src/widget/image.rs
Comment (mockersf) [inline crates/bevy_ui/src/widget/image.rs:74]:
```suggestion
        // If only one of width or height was determined at this point, then the other is set beyond this point using the aspect ratio.
```
this feels more explicit

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**
